pkg/awsutils/awsutils.go

```diff
@@ -1144,19 +1150,19 @@ func (cache *EC2InstanceMetadataCache) AllocIPAddress(eniID string) error {
 	return nil
 }
 
-// GetENIipLimit return IP address limit per ENI based on EC2 instance type
-func (cache *EC2InstanceMetadataCache) GetENIipLimit() (int, error) {
-	ipLimit, ok := InstanceIPsAvailable[cache.instanceType]
+// GetENIIPv4Limit return IP address limit per ENI based on EC2 instance type
+func (cache *EC2InstanceMetadataCache) GetENIIPv4Limit() (int, error) {
+	eniLimits, ok := InstanceNetworkingLimits[cache.instanceType]
 	if !ok {
 		log.Errorf("Failed to get ENI IP limit due to unknown instance type %s", cache.instanceType)
```

Comment:
This code doesn't fallback to `DescribeInstanceTypes` ?
Do we want to add that in this PR or a followup? (I'm ok with either - this PR doesn't make it _worse_)

Reply:
Well, this is really a bit hacky, but in `GetENILimit()` we add the missing instance type to the `InstanceNetworkingLimits` map, so as long as that function is called first, it always works. That said, we should probably move that part out to a separate function. I'll make a follow up PR.